subscription.go

```diff
@@ -634,14 +634,14 @@ func (sc *SubscriptionClient) doRaw(query string, variables map[string]interface
 	// if the websocket client is running and acknowledged by the server
 	// start subscription immediately
 	ctx := sc.getContext()
-	if ctx != nil && ctx.GetAcknowledge() {
+	if ctx != nil && sc.getClientStatus() == scStatusRunning && ctx.GetAcknowledge() {
```

Comment:
Thanks!
Taken that we do not lock, there are still chances that `Subscribe` will be triggered on a closed connection. 
But this seems correct to me, based on existing logic.

Reply:
I haven't found any absolute approach to solve this race condition issue. However, the chance is low except the user tries to close and run repeatedly in parallel. IMO it's better to create a new client and subscribe because all subscriptions were cleared after the client was closed. There isn't any benefit over the new client.